grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractUnframedGrpcService.java

```diff
@@ -182,6 +182,12 @@ static void deframeAndRespond(ServiceRequestContext ctx,
         final HttpHeaders trailers = !grpcResponse.trailers().isEmpty() ?
                                      grpcResponse.trailers() : grpcResponse.headers();
         final String grpcStatusCode = trailers.get(GrpcHeaderNames.GRPC_STATUS);
+        try {
+            requireNonNull(grpcStatusCode, "grpc-status header must exist");
+        } catch (NullPointerException e) {
+            res.completeExceptionally(e);
```

Comment:
Let's close a pooled object (a reference counting object of Netty) before returning it. The ownership of the object is passed to this method. We should remove it if we don't ship the object to another place. https://netty.io/wiki/reference-counted-objects.html
```java
PooledObjects.close(grpcResponse.content());
```

Reply:
The intention was to pass a NPE with a formatted exception message instead of manually passing one.
would this be better?
```kotlin
if (grpcStatusCode == null) {
    PooledObjects.close(grpcResponse.content());
    res.completeExceptionally(new NullPointerException("grpcStatusCode must not be null"))
    return;
}
```

Reply:
> The intention was to pass a NPE with a formatted exception message instead of manually passing one

We can do the same thing with the `if` statement. 😉 
Becaue it throws only NPE and catch it, I think there's no need to use try with catch block. 😉 

> would this be better?

Yes, it is. 😉

Reply:
Note: It's actually possible for clients to trigger this warning although there aren't any misconfigurations from server-side. I'll upload a PR so that it gets included before the next release.

https://github.com/line/armeria/blob/7741676ab15a0b93297784e87119bc5f64c3f234/grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java#L210-L212